pom.xml

```diff
@@ -1031,6 +1054,42 @@
         </plugins>
       </build>
     </profile>
+    <!-- Choose which hbase compatibility module we compile, test, and shade against.
+    Shouldn't have an effect on the generated code and artifacts apart from the contents of the the shaded jars
+    When specifying a non-default compatibility module, you also have to specify a
+    compatible hbase.version i.e mvn clean install -P phoenix-hbase-compat-2.2 -Dhbase.version=2.2.3
+    -->
+    <profile>
+      <id>phoenix-hbase-compat-2.0</id>
+      <activation>
+        <activeByDefault>true</activeByDefault>
```

Comment:
I thought this didn't actually work, but poking at it, maybe it does? I was thinking that there was some weirdness around the activeByDefault not doing what you are intending. Using `mvn help:active-profiles` looks liek the appropriate profiles are getting activated so maybe I just think life has to be harder than it needs to be ;)

Reply:
Talked to Busbey about this because I figured he would be the only other Maven guru I know who remembers this. He pointed me back to https://issues.apache.org/jira/browse/MNG-4917

The question is why, when I specified a profile, did I not see these semantics? Sounds like https://issues.apache.org/jira/browse/MNG-6787 might be affecting my 3.6.3 Maven version.

I do think this needs to be changed from 2 profiles activating 2 states to 3 profiles activating 2 states (as dumb as that is):
* profile1 activated by `! hbase.compat.version` which selects the hbase 2.0 compat module
* profile 2 activated by `hbase.compat.version=2.0` which selects the hbase 2.0 compat module
* profile 3 activated by `hbase.compat.version=2.2` which selects the hbase 2.2 compat module

Reply:
Right now I am using `hbase.compat.version` for two things
- selecting the right profile
- building the filenames and classifiers.

So the build is not designed to work correctly if you directly specify the profile.

We could probably make that work, but we'd have to push down profile handling to each module and set the filenames/classifiers directly, which would be rather ugly. 

Actually, MNG-6787 even helps us, as it means that you we can specify other profiles (namely release), and still get the implicit 2.0.1 profile as well.

However, to make behaviour more consistent with a future maven release where MNG-6787 is fixed, I agree that adding a third entry is the best option.

Reply:
I've just realized that the example in the comment is bad (uses -P ). I'll fix that tomorrow.

Reply:
fixed in last version